index.js

```diff
@@ -1,46 +1,93 @@
-async function setupPlugin({ config }) {
-    const authResponse = await fetchWithRetry('https://api.sendgrid.com/v3/marketing/contacts/count', {
+async function setupPlugin({ config, global }) {
+    // With this call we validate the API Key and also we get the list of custom fields, which will be needed
+    // to configure the map between PostHog and Sendgrid.
+    const fieldsDefResponse = await fetchWithRetry('https://api.sendgrid.com/v3/marketing/field_definitions', {
         headers: {
             Authorization: `Bearer ${config.sendgridApiKey}`
         }
     })
-
-    if (!statusOk(authResponse)) {
+    if (!statusOk(fieldsDefResponse)) {
         throw new Error('Unable to connect to Sendgrid')
     }
+
+    const fieldsDef = await fieldsDefResponse.json()
+
+    // Custom fields in Sendgrid have a name and an ID. The name is what users configure when they create a custom field,
+    // and ID is automatically assigned by Sendgrid.
+    // In the config of this plugin, users configure the map between PostHog prop names and Sendgrid custom fields names.
+    // Here we resolve the relation and calculate a map between PostHog prop names and Sendgrid custom field IDs.
+
+    let posthogPropsToSendgridCustomFieldNamesMap = {}
+    try {
+        posthogPropsToSendgridCustomFieldNamesMap = parseCustomFieldsMap(config.customFields)
+    } catch (e) {
+        throw new Error('Invalid format for custom properties: ' + e)
```

Comment:
We're introducing plugin logs soon https://github.com/PostHog/posthog/pull/3482 so I'd actually leverage that in this plugin. 

I'd log the exception `e` and throw the error with the simplified message, as that will be most useful for non-technical users

Reply:
So if I understand the PR correctly, I just have to `console.error("content of the error")` before throwing the exception. Am I correct?

Reply:
If so, I have implemented that in 7e0ef9f.